pipeline-notebooks/pipeline-general.ipynb

```diff
@@ -540,19 +551,44 @@
    "outputs": [],
    "source": [
     "# pipeline-api\n",
-    "def pipeline_api(file, filename='', response_type=\"application/json\"):\n",
+    "\n",
+    "class UnsupportedFileError(Exception):\n",
+    "    def __init__(self, message):\n",
+    "        self.message = message\n",
+    "        super().__init__(self.message)\n",
+    "\n",
+    "DEFAULT_MIMETYPES = \"application/pdf,application/msword,image/jpeg,image/png,text/markdown,\" \\\n",
+    "                    \"text/x-markdown,application/epub,application/epub+zip,text/html,\" \\\n",
+    "                    \"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,\" \\\n",
+    "                    \"application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.\" \\\n",
+    "                    \"presentationml.presentation,\" \\\n",
+    "                    \"application/vnd.ms-powerpoint,application/xml\"\n",
+    "\n",
+    "\n",
+    "if not os.environ.get(\"UNSTRUCTURED_ALLOWED_MIMETYPES\", None):\n",
+    "    os.environ[\"UNSTRUCTURED_ALLOWED_MIMETYPES\"] =  DEFAULT_MIMETYPES\n",
+    "\n",
+    "\n",
+    "def pipeline_api(file, filename='', file_content_type=None, file_filename=\"\", response_type=\"application/json\"):\n",
     "    # NOTE(robinson) - This is a hacky solution due to\n",
     "    # limitations in the SpooledTemporaryFile wrapper.\n",
-    "    # Specifically, it does't have a `seekable` attribute,\n",
+    "    # Specifically, it doesn't have a `seekable` attribute,\n",
     "    # which is required for .pptx and .docx. See below\n",
     "    # the link below\n",
     "    # ref: https://stackoverflow.com/questions/47160211\n",
     "    # /why-doesnt-tempfile-spooledtemporaryfile-implement-readable-writable-seekable\n",
+    "    if file_content_type and file_content_type not in os.environ.get(\"UNSTRUCTURED_ALLOWED_MIMETYPES\"):\n",
+    "        raise UnsupportedFileError(message=f\"{file_content_type} not currently supported\")\n",
```

Comment:
this should already be taken care of by the unstructured-api-tools generated code, e.g. https://github.com/Unstructured-IO/unstructured-api-tools/blob/060ead4/test_unstructured_api_tools/pipeline-test-project/prepline_test_project/api/process_file_1.py#L60